src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -387,4 +423,138 @@ private PreparedStatement getPreparedStatement(
     private Iterable<MapValue> doQuery(QueryRequest qReq) {
         return new IterableUtil.IterableImpl(nosqlClient, qReq);
     }
+
+    Map<String, FieldValue.Type> getShardKeys(Field idField) {
```

Comment:
Can't this method and getNonShardKeys() be moved into NosqlEntityInformation?

Reply:
No. We need mappingNosqlConverter and its not accessible from NosqlEntityInformation